deployment/Triton/client/client.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from tritonclient.utils import *
+import tritonclient.grpc as grpcclient
+import tritonclient.http as httpclient
+
+import argparse
+import numpy as np
+import os
+import sys
+import time
+from uuid import uuid4
+
+model_name = "monai_covid"
+
+
+def open_nifti_files(input_path):
+
+    nifti_file_names = []
+
+    for root, dirs, files in os.walk(input_path):
+        for file in files:
+            if file.endswith('.nii') or file.endswith('.nii.gz'):
+                nifti_file_names.append(os.path.join(input_path, file))
+
+    return nifti_file_names
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description='Triton CLI for COVID classification inference from NIFTI data')
+    parser.add_argument(
+        'input',
+        type=str,
+        help="Path to NIFTI file or directory containing NIFTI files to send for COVID classification"
+    )
+    args = parser.parse_args()
+
+    nifti_files = []
+    if os.path.isdir(args.input):
+        nifti_files = open_nifti_files(args.input)
+    elif os.path.isfile(args.input):
+        nifti_files = [args.input]
+
+    if not nifti_files:
+        print("No valid inputs provided")
+        sys.exit(1)
+
+    with httpclient.InferenceServerClient("localhost:8000") as client:
+        image_bytes = b''
+        for nifti_file in nifti_files:
+            with open(nifti_file, 'rb') as f:
```

Comment:
are u looking for a generic wrapper here for triton client.. that can be useful to trigger infer requests for simple inferer or sliding window inferer?

Reply:
Great comments. It is just for demo purposes.  In fact, we provide a shell script that utilizes this code to read three actual lung XR nifti files, runs the transforms and then outputs the inference. 
We plan to enhance the read method in later updates and welcome an issue request for that.
See the Readme.md and Quick Start section for our example of reading something, running the transforms and sending the input to triton.  The code is included.

Reply:
There are a few ways to use Triton as an inference server
1. Load TorchScript model only on Triton server (model file and the config.pbxtxt), and use Triton KFServing predict API, via Triton Client SDK. This way, all the data transforms would be on the client side (including loading, resampling, etc) and the MONAI SimpleInferer or SlidingWindowinferer will be using a specialized network/predictor for Triton (built with Triton Client SDK). Flowing between the Client and Server are the actual input tensors and output tensor for the model. I think this is what Sachi was asking. AIAA and Clara Deploy has been using this approach for Triton inference.     
2. Wrap the whole application as a Triton Python backend, and use it as if a model in Triton Server. This way, the data transforms (loading, resampling etc), the actual model loading, and inference are done in the Python backend itself. The client side then will only send over the bytes of the input file (NIfTI, MHD, or others) through the Triton KFServing API using Triton Client SDK. The code in review is taking this approach, and I believe for simplicity, Triton Docker image was also used as the base to add the Python backend code and build the demo Docker. In a multi-model serving setup, a Python backend, like other models, would be provided to a Triton docker via local file system or cloud storage.